File: mukha/lab_3/PortManagementSystem/factories/ship_factory.py
from ships.ship import Ship
from ships.light_weight_ship import LightWeightShip
from ships.medium_ship import MediumShip
from ships.heavy_ship import HeavyShip

class ShipFactory:
    def create_ship(self, ship_type, config):
        if ship_type == "LightWeight":
            return LightWeightShip(config)
        elif ship_type == "Medium":
            return MediumShip(config)
        elif ship_type == "Heavy":
            return HeavyShip(config)
        else:
            return Ship(config)

File: dmytryshyn/Lab3/Lab3/Lab3/Classes/Ships/ShipFactory.cs
using Lab3.Classes.Ships.Implementations;
using Lab3.Enums;
using MySecondPattern.Classes.Ports;
using MySecondPattern.Classes.Ships;
using MySecondPattern.Classes.Ships.Characteristics;


namespace Lab3.Classes.Ships
{
    public static class ShipFactory
    {
        public static Ship CreateShip(ShipTypes shipType, int shipId, ShipLimitsCharacteristics shipConfig, double fuel)
        {
            switch (shipType)
            {
                case ShipTypes.LightWeightShip:
                    return new LightWeightShip(shipId, fuel, shipConfig);
                case ShipTypes.MediumShip:
                    return new MediumShip(shipId, fuel, shipConfig);
                case ShipTypes.HeavyShip:
                    return new HeavyShip(shipId, fuel, shipConfig);
                default:
                    throw new ArgumentException($"Invalid item type: {shipType}");
            }
        }
    }
}
